src/sentry/uptime/detectors/ranking.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+from datetime import timedelta
+from random import random
+from typing import TYPE_CHECKING
+
+from django.conf import settings
+from redis.client import StrictRedis
+from rediscluster import RedisCluster
+
+from sentry.utils import redis
+
+if TYPE_CHECKING:
+    from sentry.models.project import Project
+
+
+# How often should each project be flushed?
+PROJECT_FLUSH_FREQUENCY = timedelta(days=1)
+# How often do we want to run our task to flush the buckets?
+# XXX: This might actually belong in the task when we have that?
+BUCKET_FLUSH_FREQUENCY = timedelta(minutes=1)
+# How often should we trim the ranked list?
+RANKED_TRIM_CHANCE = 0.01
+# How many urls should we trim the ranked list to?
+RANKED_MAX_SIZE = 20
+# Expiry we should set on all keys. Set the expiry to twice the flush frequency
+# so that we definitely have time to process.
+KEY_EXPIRY = PROJECT_FLUSH_FREQUENCY * 2
+
+
+def _get_cluster() -> RedisCluster | StrictRedis:
+    return redis.redis_clusters.get(settings.SENTRY_UPTIME_DETECTOR_CLUSTER)
+
+
+def add_hostname_to_rank(project: Project, hostname: str):
```

Comment:
What is the reason we rank them?

Reply:
We want to find the most common hostnames in a project and attempt to monitor the top one.

Reply:
Ahh right 👍